x.payment_batching/payment_batching.md

```diff
@@ -0,0 +1,307 @@
+procedure.  In the first step, multiple small inputs are
+[consolidated][chapter consolidation] into a single larger input using
+slow (but low-feerate) transactions that spend the service's money back
+to itself.  In the second step, the service spends from one of its
+consolidated inputs using payment batching and achieves the best-case
+efficiency described above.
+
+If we assume that consolidation transactions will pay only 20% of the
+feerate of a normal transaction and will consolidate 100 inputs at a
+time, we can calculate the savings of using the two-step procedure for
+our 10-to-1 and 1-to-1 scenarios above (while showing, for comparison,
+the simple best-case scenario of already having a large input available).
+
+![Saving rates for best, typical, and tough cases of payment batching after consolidation](img/p2wpkh-batching-after-consolidation.png)
+
+Most notably, the savings from consolidation allows the tough case go
+from worst performing to best performing.  For the typical case,
+consolidation actually loses money when only making a single payment,
+but when actually batching, it performs almost as well as the best case
+scenario.
+
+In addition to payment batching directly providing a fee savings,
+batching also uses limited block space more efficiently by reducing the
+number of vbytes per payment.  This increases the available supply of
+block space and so, given constant demand, can make it more affordable.
+In that way, increased use of payment batching may lower the feerate for
+all Bitcoin users.
+
+In summary, payment batching provides significant savings for services
+that typically have inputs available that are 5 to 20 times larger than
+their typical output.  For services not in that position, the savings
+from batching alone are smaller but perhaps still worth the effort;
+if the services are willing to also pre-consolidate their inputs, the
+savings can be quite dramatic.
+
+Note: the figures and plots above all assume use of P2WPKH inputs and
+outputs.  We expect that to become the dominant script type on the
+network in the future (until something better comes along).  However, if
+you use a different script type (P2PKH, or multisig using P2SH or
+P2WSH), the number of vbytes used to spend them are even larger, so the
+savings rate will be higher.
+
+## Problems
+
+The fee-reduction benefits of payment batching do create tradeoffs and
+problems that will need to be addressed by any service using the
+technique.
+
+### Delayed gratification
+
+This is the primary problem with payment batching.  Although some
+situations naturally lend themselves to payment batching (e.g.  sending
+out the company payroll for every employee at the same time), many
+services primarily send money to users when those users make a
+withdrawal request.  In order to batch payments, the service must get
+the user to accept that their payment will not be sent immediately---it
```

Comment:
Although you're correct that batching will not create a delay in receiving 1-confirmation payments, it'll still be a delay for users who expect notification of unconfirmed transactions (unless they using rolling RBF updates).  See also this old tweet of mine for some color: https://twitter.com/hrdng/status/739825949466845185

Reply:
This does actually introduce a delay, even for frequent spending. We always expect the first block to be mined _10 minutes after a transaction is transmitted_. That means that if an exchange holds on to a withdraw orderfor five minutes to batch with other payments before sending, then the first expected block is 15 minutes after the order is received. There's no way to avoid this - every second you hold on to a transaction before transmitting adds a second to the expected time that it'll be mined.

The difference is that if an exchange is sending out payments frequently, it can benefit a bit more by batching:

1. if it's sending a payment every minute, it could send a batch every ten minutes. No payment would have more than 10 minutes additional delay added, but on average, the exchange would save 75% in the best case.
2. if it's sending a payment every thirty minutes and it 'batched' every ten minutes, then the payment would have 10 minutes of delay added, but on average each transaction would only contain one payment, so no savings would be made.

We can take this further. If Bitcoin blockspace does become competitive, then frequent spenders could use batching not to save fees, but to increase priority (by attaching the same total fee as would have been used without batching, the feerate increases). In that model, there would be an introduced delay before the transaction is broadcast, but we would expect it to confirm in fewer blocks.

Reply:
Good stuff! I think it'd be valuable to add to the chapter.